Issue 212: Updates to User Profile

twistedstream opened the pull request:
Starting a PR to collaborate on the updates to User Profile described in #212, with a special heads-up to @yvonnewilson and @eugeniop.

Closes #212

twistedstream committed:
Move user-profile.md to new user-profile/ dir

as normalized.md
user-profile/index.md will become the new page when someone clicks "User Profile" in the left nav

twistedstream committed:
Port of existing draft of User Profile doc

to user-profile/index.md

yvonnewilson committed:
new document with overview of token info

yvonnewilson committed:
Fix missing backticks for code sample

yvonnewilson committed:
Cleaned up comments, added ticks around id_token

yvonnewilson committed:
Update tokens.md

twistedstream commented on articles/tokens.md:
```diff
@@ -0,0 +1,484 @@
+Tokens should be issued with a short timeframe for expiration, where configurable.
+
+
+## The Auth0 APIv2 access token
```
@yvonnewilson, I found in [this section](https://auth0.com/docs/api/v2/changes#7) of the **API v1 vs v2** docs page that it calls this token the `api_jwt_token`.  What's nice about this term is that it gives the Auth0 APIv2 access token a similar moniker as the other tokens like `id_token` and `access_token`.  Should we consider using this in our docs?

yvonnewilson replied:
@twistedstream Hi, I like the 'api' part of the name, but not sure we need the 'jwt' in it.  None of the other names allude to internal format details of the token.  I think it would be more helpful to clarify in the name which api it is for because there are several apis, each of which requires a different token.  It would eliminate a lot of confusion if
a) we use the same name (fully qualified) consistently throughout our docs for each type of token (e.g. don't just call it 'the token')
b) each token is named to indicate what it is and/or what you use it for. - auth0 api token, IDP api token, 3rd party app API token, etc.  
I'm not strongly against having jwt in the name, but I think the above 2 things are more important in helping people understand all these tokens and which ones they need.

yvonnewilson committed:
Remove all editorial notes about additional things to add

Removed all editorial notes about additional things to add so we can at least publish this content so far

yvonnewilson committed:
Removed one last editorial comment

Removed one last editorial comment

yvonnewilson committed:
Convert all URLs to new format

Made all URL references relative, where possible

yvonnewilson committed:
Remove editorial comment

Remove editorial comment

yvonnewilson committed:
Fix reference to sequence diagrams

Fixed one sequence diagram reference missed earlier

twistedstream committed:
Merge branch 'master' into issue-212_updates-to-user-profile

ntotten commented:
@twistedstream Whats the status on this?

twistedstream committed:
Merge branch 'master' into issue-212_updates-to-user-profile

twistedstream committed:
Final tweaks to main user profile doc

including the addition of a TOC

twistedstream committed:
Tweaks to Normalized User Profile

twistedstream committed:
Move tokens.md to /tokens/index.md

and make Tokens a member of the How To's group

twistedstream committed:
Break tokens/index.md into separate files

for each token type, with links to them from index.md

twistedstream commented:
@ntotten This PR is ready for review and merge. 

It includes new User Profile content as well as a new section that describes the various types of tokens (JWT, access_token, refresh_token, etc) used by Auth0, with a new entry in `config/navigation.yml` so Tokens shows up in the **How To's** section. It's not perfect, but it's a good start. @yvonnewilson and @eugeniop wanted to get this in sooner than later.

We'd like to leave the branch open so we can continue to revise content and add things like illustrations.

twistedstream committed:
Merge branch 'master' into issue-212_updates-to-user-profile

twistedstream committed:
Remove Tokens from How To's menu

and provide a stronger link to it from the User Profile page

twistedstream committed:
Add links to Tokens document

from other existing token docs
